modules/ROOT/pages/advanced-mule-concepts.adoc

```diff
@@ -0,0 +1,17 @@
+= Advanced Mule Concepts
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: advanced, mule, concepts, runtime, engine
+
+This section covers some advanced Mule topics that will be useful once you are familiar with the basics of Mule application development and Mule runtime engine concepts.
+
+Learn about the xref:execution-engine.adoc[Execution Engine] to understand how the Mule runtime engine processes the operations in a Mule application’s flow, and understand how to change the thread pools if needed. Mule runtime engine uses execution metadata provided by operations in a flow to perform auto-tuning.
+
+The xref:about-classloading-isolation.adoc[Classloading Isolation] mechanism implemented in Mule 4 addresses potential issues caused when conflicting versions of the same library are visible in the same location.
```

Comment:
the original motivation of classloading isolation is to make upgrades easier by not allowing a component to expose its internals which may change, and components to interact between themselves through a well defined, complete and as-minimal-as-possible api.